### ISSUE TITLE
 reproduce experiment on ACDC dataset

### ISSUE THREAD

**Issue by xhu248** [LiheYoung/UniMatch#83]
Thank you for this great work. I just used my own created split (in my previous works) rather than the one provided by the author. After training for 300 epochs, the dice scores using 1 case and 5 cases are both around 20, far lower than 86 reported in this paper. I couldn't find a reason for it. did anyone ever succeed in reproducing the results on ACDC dataset?

**Comment by LiheYoung** [comment]
Hi, have you tried our provided splits?

**Comment by xhu248** [comment]
![image](https://github.com/LiheYoung/UniMatch/assets/45184617/6215a42a-f847-4d81-a48b-7dbba37e0381)
i'm trying the split you provided, and the setting is 1 case

here is the log you provided for the first epoch
![image](https://github.com/LiheYoung/UniMatch/assets/45184617/f0c310be-3d10-47cf-a830-50ddd4e6dd32)

here is the log of my reproduction:
![image](https://github.com/LiheYoung/UniMatch/assets/45184617/f49f30a6-0c75-4864-bd1d-2f0cc9ecaa58)


ill wait for 300 epochs and get back to you later

**Comment by xhu248** [comment]
hi, i've tried your provided split and finished training, here are the results

val dice score:
![image](https://github.com/LiheYoung/UniMatch/assets/45184617/974816d1-63b9-446a-b51e-1aba02897ddb)

training loss:
![image](https://github.com/LiheYoung/UniMatch/assets/45184617/abe9482c-6fab-460b-811c-71dadc70aced)

the final dice score is around **15.48** and best dice score is **17.01**

**Comment by LiheYoung** [comment]
Hi, thank you for your feedback. I will re-run and re-check the code these two days. Could you upload the training log as well? Thank you.

**Comment by xhu248** [comment]
thank you for your response, here is the log file
[20230920_025111.log](https://github.com/LiheYoung/UniMatch/files/12707624/20230920_025111.log)

**Comment by LiheYoung** [comment]
Sorry for the late reply. I have re-run my code, and achieved similar results as our paper on all settings. Below are the training logs, tensorboards, and checkpoints. Please check if you have modified our code.
| Labeled cases | 1 | 3 | 7 |
|:-:|:-:|:-:|:-:|
| Results | 85.5 | 88.7 | 89.5 |
| Download | [log](https://drive.google.com/file/d/1wGLyrOPEMMTlzv0wWjDsnoTITr3JqOdc/view?usp=sharing) \| [tensorboard](https://drive.google.com/file/d/1IHmaq9zA56H_n2ShPOcEyPn-fpB5z05b/view?usp=sharing) \| [ckp](https://drive.google.com/file/d/1k9cUi2b21jWb7bjmB-TKUknDNodrQu4M/view?usp=sharing) | [log](https://drive.google.com/file/d/1OuXr9K3Uu8pdUSS84713tAIKqxeV5W6e/view?usp=sharing) \| [tensorboard](https://drive.google.com/file/d/1Qis9oGqJ0PJAvG_TpDou-DVc0doxxWOs/view?usp=sharing) \| [ckp](https://drive.google.com/file/d/1XjQgqbfsTLe6tQWK3i7uQBcQsUuvWnul/view?usp=sharing) | [log](https://drive.google.com/file/d/16j5degLX2RGP0o2kl4zYFHgLiVTenGtS/view?usp=sharing) \| [tensorboard](https://drive.google.com/file/d/1YnbNX7NeTrOb5uhBW0hQ_NZ7Q3O3yLtl/view?usp=sharing) \| [ckp](https://drive.google.com/file/d/1QZJm5a2gmBG_3zuNnoKh5pmQI3HQQCq6/view?usp=sharing) |

**Comment by xhu248** [comment]
thank you for your reply, i checked the codes, the only part i modified is the dataloader (as a fair comparison, and i cannot download the data from SSL4MIS as the BAIDU clouddisk link didn't work out), which shouldn't have contributed to such a huge gap between your results and mine. 

idk, maybe we can see if others can reproduce the experiments on ACDC dataset.

Thank you again, appreciate your time for rerunning all experiments.

**Comment by lichen14** [comment]
@LiheYoung Hi Young, thanks for your work.
I am trying to reproduce the result on ACDC dataset.
I have found your [training-logs](https://github.com/LiheYoung/UniMatch/blob/0324f3dd4a22128105a5eb135d2586c0bb3cc647/more-scenarios/medical/training-logs/3cases.log#L8C22-L8C51) and wonder where is the  'labeled_id_path': 'partitions/acdc/3/labeled.txt', 'unlabeled.txt'?
Similar, the 'partitions/acdc/1/(un)labeled.txt' and 'partitions/acdc/7/(un)labeled.txt'.
Can you provide the dataset split?

Thanks,
Chen

**Comment by lichen14** [comment]
> @LiheYoung Hi Young, thanks for your work. I am trying to reproduce the result on ACDC dataset. I have found your [training-logs](https://github.com/LiheYoung/UniMatch/blob/0324f3dd4a22128105a5eb135d2586c0bb3cc647/more-scenarios/medical/training-logs/3cases.log#L8C22-L8C51) and wonder where is the 'labeled_id_path': 'partitions/acdc/3/labeled.txt', 'unlabeled.txt'? Similar, the 'partitions/acdc/1/(un)labeled.txt' and 'partitions/acdc/7/(un)labeled.txt'. Can you provide the dataset split?
> 
> Thanks, Chen

Forget my stupid question above. Looks like I've found the relevant files, [here](https://github.com/LiheYoung/UniMatch/tree/0324f3dd4a22128105a5eb135d2586c0bb3cc647/more-scenarios/medical/splits/acdc).
I'll report on any future updates on the reproduction results as well.

**Comment by lichen14** [comment]
Hi @xhu248, I just tried the split provided, and the setting was 3 case.
The UniMatch worked well on ACDC dataset and obtained similar results. (88.7 v.s. my 87.58)
In my exp, the Dice under 3 labelled cases was 87.58% after 300 epochs.
Here is the log.
[20241029_103523.log](https://github.com/user-attachments/files/17551196/20241029_103523.log)
